packages/storage/src/providers/s3/apis/getProperties.ts

```diff
@@ -1,5 +1,64 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-// TODO: pending implementation
-export declare const getProperties: (params: any) => Promise<any>;
+import { HeadObjectInput, headObject } from '../../../AwsClients/S3';
+import { StorageOptions, StorageOperationRequest } from '../../../types';
+import { assertValidationError } from '../../../errors/utils/assertValidationError';
+import { StorageValidationErrorCode } from '../../../errors/types/validation';
+import { GetPropertiesException, S3GetPropertiesResult } from '../types';
+import {
+	resolveStorageConfig,
+	getKeyWithPrefix,
+	resolveCredentials,
+} from '../utils';
+
+/**
+ * Get Properties of the object
+ *
+ * @param {StorageOperationRequest} The request object
+ * @return {Promise<S3GetPropertiesResult>} Properties of the object
+ * @throws service: {@link GetPropertiesException}
+ * - S3 Service errors thrown while getting properties
+ * @throws validation: {@link StorageValidationErrorCode } - Validation errors thrown
+ *
+ */
```

Comment:
Great ideas to add throws here, thanks.
One suggestion on the formate:

Probably need to verify the format of the added comment docs. E.g. the doc of `getProperties` looks like this currently:

<img width="584" alt="image" src="https://github.com/aws-amplify/amplify-js/assets/10602282/d80624b2-5588-4da6-be51-d5929b785fe3">

```suggestion
/**
 * Gets the properties of a file. The properties include S3 system metadata and
 * the user metadata that was provided when uploading the file.
 *
 * @param {StorageOperationRequest} req The request to make an API call.
 * @returns {Promise<S3GetPropertiesResult>} A promise that resolves the properties.
 * @throws A {@link GetPropertiesException} when the underlying S3 service returned error.
 * @throws A {@link StorageValidationErrorCode} when API call parameters are invalid.
 */
```

<img width="561" alt="image" src="https://github.com/aws-amplify/amplify-js/assets/10602282/64ee92d9-e2eb-4706-9605-a4809bde5631">

Please check the accuracy other the inline-docs else where, too.